src/utils/utf8.c

```diff
@@ -152,29 +182,53 @@ utf8_toupper(utf8_ctx_t *ctx, const char *s, char *buf, size_t buflen)
 	return len;
 }
 
+
+ssize_t
+utf8_subs_diacritics(utf8_ctx_t *ctx, UChar *buf, ssize_t buflen)
+{
+	UErrorCode ec = U_ZERO_ERROR;
+	int32_t textLim = (int32_t)buflen;
+	int32_t len = (int32_t)buflen;
+	
+	utrans_transUChars(
+		ctx->transliterator, buf, &len, 
+		MAX_BUFFER_SIZE, 0, &textLim, &ec
+	);
+
+	if (__predict_false(U_FAILURE(ec) || len == buflen)) {
+		return -1;
+	}
+
+	return len;
+}
+
 /*
  * utf8_normalize: lowercase the UTF-8 string and normalize using
  * Normalization Form KC (NFKC).
  *
  * See: https://www.unicode.org/reports/tr15/
+ * 
  */
 ssize_t
 utf8_normalize(utf8_ctx_t *ctx, strbuf_t *buf)
 {
 	UErrorCode ec = U_ZERO_ERROR;
-	uint16_t buf1[128], buf2[128]; // FIXME
+	uint16_t buf1[MAX_BUFFER_SIZE], buf2[MAX_BUFFER_SIZE]; // FIXME
 	ssize_t len;
 
 	if (utf8_to_utf16(ctx, buf->value, buf1, __arraycount(buf1)) == -1) {
 		return -1;
 	}
 	len = unorm2_normalize(ctx->normalizer,
 	    buf1, -1 /* NUL-terminated */, buf2, __arraycount(buf2), &ec);
+
 	if (__predict_false(U_FAILURE(ec) || (size_t)len == __arraycount(buf2))) {
 		return -1;
 	}
 	if (strbuf_prealloc(buf, len + 1) == -1) {
 		return -1;
 	}
+	utf8_subs_diacritics(ctx, buf2, len);
```

Comment:
This should go here: https://github.com/rmind/nxsearch/blob/main/src/core/filters_builtin.c#L66